packages/loader/container-loader/src/container.ts

```diff
@@ -748,7 +746,14 @@ export class Container
 		// Prefix all events in this file with container-loader
 		this.mc = loggerToMonitoringContext(ChildLogger.create(this.subLogger, "Container"));
 
-		this.options = cloneDeep(this.loader.services.options);
+		// NOTE: stringify + parse is usually bad practice. Here we deliberately chose to use it because we know the
+		// pieces we care about in the cloned object are primitive values, and it lets us avoid importing a full
+		// deep-clone library, so we keep the bundle size down. This code doesn't run a ton of times and if it's slower
+		// than an actual deep clone (improbable), the gain in network transfer time from a smaller bundle almost certainly
+		// outweighs that. The dep is still there but the cloneDeep module of lodash gets tree-shaken so we still get the
+		// bundle size improvement.
+		// Tracking alternative ways to handle this in AB#4129.
+		this.options = JSON.parse(JSON.stringify(this.loader.services.options));
```

Comment:
Could `this.loader.services.options` be `undefined`?  If so this will throw.

Reply:
Checking and patching for the edge case

Reply:
Doesn't look like it should but handled the case anyway

Reply:
According to types it can't be, so I think it's ok to leave it.  There's something to be said for defense in depth though, but generally we should assume callers will abide by the types.